opcua.cpp

```diff
@@ -1418,6 +1511,12 @@ SOPC_ReturnStatus OPCUA::initializeS2sdk(const char *traceFilePath)
 			throw runtime_error("Unable to initialise S2OPC ClientHelper library");
 		}
 
+		// Set asynchronous response callback
+		if (initStatus == SOPC_STATUS_OK)
+		{
+			initStatus = SOPC_ClientConfigHelper_SetServiceAsyncResponse(asyncS2ResponseCallBack);
```

Comment:
Throw a _runtime_error_ after this line as is done with all of the other initialization steps. The _initializeS2sdk_ method is called in two places but the return code is not checked in either one. On the other hand, the _try...catch_ implementation is fine.

Reply:
To be consistent with the other two initialization steps, add a _Logger::getLogger()->fatal(...)_ before throwing. I understand the message will be written either way.